tutorials/optimization/cpu_optimization.rst

```diff
@@ -0,0 +1,258 @@
+CPU will be waiting around for data instead of doing any work. Instead, if you
+can make your data accesses localised, or even better, access memory in a linear
+fashion (like a continuous list), then the cache will work optimally and the CPU
+will be able to work as fast as possible.
+
+Godot usually takes care of such low-level details for you. For example, the
+Server APIs make sure data is optimized for caching already for things like
+rendering and physics. But you should be especially aware of caching when using
+GDNative.
+
+Languages
+=========
+
+Godot supports a number of different languages, and it is worth bearing in mind
+that there are trade-offs involved - some languages are designed for ease of
+use, at the cost of speed, and others are faster but more difficult to work
+with.
+
+Built-in engine functions run at the same speed regardless of the scripting
+language you choose. If your project is making a lot of calculations in its own
+code, consider moving those calculations to a faster language.
+
+GDScript
+~~~~~~~~
+
+GDScript is designed to be easy to use and iterate, and is ideal for making many
+types of games. However, ease of use is considered more important than
+performance, so if you need to make heavy calculations, consider moving some of
+your project to one of the other languages.
+
+C#
+~~
+
+C# is popular and has first class support in Godot. It offers a good compromise
+between speed and ease of use.
+
+Other languages
+~~~~~~~~~~~~~~~
+
+Third parties provide support for several other languages, including `Rust
+<https://github.com/godot-rust/godot-rust>`_ and `Javascript
+<https://github.com/GodotExplorer/ECMAScript>`_.
+
+C++
+~~~
+
+Godot is written in C++. Using C++ will usually result in the fastest code,
+however, on a practical level, it is the most difficult to deploy to end users'
+machines on different platforms. Options for using C++ include GDNative, and
+custom modules.
+
+Threads
+=======
+
+Consider using threads when making a lot of calculations that can run parallel
+to one another. Modern CPUs have multiple cores, each one capable of doing a
+limited amount of work. By spreading work over multiple threads you can move
+further towards peak CPU efficiency.
+
+The disadvantage of threads is that you have to be incredibly careful. As each
+CPU core operates independently, they can end up trying to access the same
+memory at the same time. One thread can be reading to a variable while another
+is writing. Before you use threads make sure you understand the dangers and how
+to try and prevent these race conditions.
+
+For more information on threads see :ref:`doc_using_multiple_threads`.
+
+SceneTree
+=========
+
+Although Nodes are an incredibly powerful and versatile concept, be aware that
+every node has a cost. Built in functions such as `_process()` and
+`_physics_process()` propagate through the tree. This housekeeping can lead to
+situations when you have very large numbers of nodes.
```

Comment:
pathological situations

Reply:
Pathological is a big word for a lot of users and will be hard to translate.

So I went with "This housekeeping can reduce performance when you have very large numbers of nodes."